rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbSupportImplTest.java

```diff
@@ -0,0 +1,238 @@
+        when(feePerKbConstants.getFeePerKbChangeAuthorizer()).thenReturn(authorizer);
+        Transaction tx = mock(Transaction.class);
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        when(authorizer.isAuthorized(tx, signatureCache)).thenReturn(true);
+
+        Coin negativeFeePerKb = mock(Coin.class);
+        when(negativeFeePerKb.isPositive()).thenReturn(false);
+
+        Integer actualResult = feePerKbSupport.voteFeePerKbChange(tx, Coin.NEGATIVE_SATOSHI, signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.NEGATIVE_FEE_VOTED.getCode();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_unsuccessfulVote_excessiveFeePerKb() {
+        final long MAX_FEE_PER_KB = 5_000_000L;
+        AddressBasedAuthorizer authorizer = mock(AddressBasedAuthorizer.class);
+        when(feePerKbConstants.getFeePerKbChangeAuthorizer()).thenReturn(authorizer);
+        Transaction tx = mock(Transaction.class);
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        when(authorizer.isAuthorized(tx, signatureCache)).thenReturn(true);
+
+        Coin feePerKb = mock(Coin.class);
+        when(feePerKb.isPositive()).thenReturn(true);
+
+        Coin maxFeePerKb = mock(Coin.class);
+        when(feePerKbConstants.getMaxFeePerKb()).thenReturn(maxFeePerKb);
+        when(feePerKb.isGreaterThan(maxFeePerKb)).thenReturn(true);
+
+        Integer actualResult = feePerKbSupport.voteFeePerKbChange(tx, Coin.valueOf(MAX_FEE_PER_KB), signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.EXCESSIVE_FEE_VOTED.getCode();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_unsuccessfulVote() {
+        final String SET_FEE_PER_KB_ABI_FUNCTION = "setFeePerKb";
+        AddressBasedAuthorizer authorizer = mock(AddressBasedAuthorizer.class);
+        when(feePerKbConstants.getFeePerKbChangeAuthorizer()).thenReturn(authorizer);
+        Transaction tx = mock(Transaction.class);
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        when(authorizer.isAuthorized(tx, signatureCache)).thenReturn(true);
+
+        Coin feePerKb = mock(Coin.class);
+        when(feePerKb.isPositive()).thenReturn(true);
+
+        Coin maxFeePerKb = spy(Coin.CENT);
+        when(feePerKbConstants.getMaxFeePerKb()).thenReturn(maxFeePerKb);
+        when(feePerKb.isGreaterThan(maxFeePerKb)).thenReturn(false);
+
+        ABICallElection feePerKbElection = spy(new ABICallElection(authorizer));
+        when(provider.getFeePerKbElection(authorizer)).thenReturn(feePerKbElection);
+
+        ABICallSpec feeVote = spy(new ABICallSpec(SET_FEE_PER_KB_ABI_FUNCTION, new byte[][]{
+            BridgeSerializationUtils.serializeCoin(feePerKb)}));
+        when(feePerKbElection.vote(feeVote, tx.getSender(signatureCache))).thenReturn(false);
```

Comment:
This can be simplified without spying objects

```suggestion
        ABICallElection feePerKbElection = mock(ABICallElection.class);
        when(feePerKbElection.vote(any(), any()).thenReturn(false);

```